service/tablet_allocator.cc

```diff
@@ -305,8 +411,79 @@ class load_balancer {
             lblogger.info("Prepared {} migrations in DC {}", dc_plan.size(), dc);
             plan.merge(std::move(dc_plan));
         }
+        plan.merge(co_await make_resize_plan());
 
-        lblogger.info("Prepared {} migrations", plan.size());
+        lblogger.info("Prepared {} migration plans, out of which, tablet_migrations={} and resize_requests={}",
+                      plan.size(), plan.tablet_migrations(), plan.resize_requests());
+        co_return std::move(plan);
+    }
+
+    future<migration_plan> make_resize_plan() {
+        migration_plan plan;
+
+        if (!_tm->tablets().balancing_enabled()) {
+            co_return std::move(plan);
+        }
+
+        cluster_resize_stats resize_load;
+
+        for (auto&& [table, tmap_] : _tm->tablets().all_tables()) {
+            auto& tmap = tmap_;
+
+            auto avg_tablet_size = _stats.load_stats_for_table(table).size_in_bytes / std::max(tmap.tablet_count(), size_t(1));
+            resize_load.update(table, avg_tablet_size, tmap.resize());
```

Comment:
The call tmap.resize() looks like we are resizing it;). Perhaps we can rename it to get_reisze().

Reply:
you're right. will rename it to resize_type or resize_decision.